src/cmd/windows/durable_task_monitor.go

```diff
@@ -0,0 +1,179 @@
+// +build windows
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+	"jenkinsci.org/plugins/durabletask/common"
+)
+
+type Shell int
+
+const (
+	CMD Shell = iota
+	POWERSHELL
+	PWSH
+)
+
+func (shell Shell) String() string {
+	switch shell {
+	case CMD:
+		return "cmd"
+	case POWERSHELL:
+		return "powershell"
+	case PWSH:
+		return "pwsh"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// Launches the script in a new session and waits for its completion.
+func launcher(wg *sync.WaitGroup, exitChan chan bool, shell string,
+	scriptPath string, resultPath string, outputPath string,
+	launchLogger *log.Logger, scriptLogger *log.Logger) {
+
+	defer wg.Done()
+	defer common.SignalFinished(exitChan)
+
+	if _, err := os.Stat(scriptPath); err != nil {
+		if os.IsNotExist(err) {
+			launchLogger.Printf("%s does not exist", scriptPath)
+			return
+		}
+	}
+
+	var scriptCmd *exec.Cmd
+	switch shell {
+	case CMD.String():
+		scriptCmd = exec.Command("cmd.exe", "/C", scriptPath)
+	case POWERSHELL.String():
+		shellCommand := fmt.Sprintf("[Console]::OutputEncoding = [Text.Encoding]::UTF8; .\\%s", scriptPath)
```

Comment:
So, if I recall, if I do not set the encoding here to `UTF-8`, I get `ANSI` for my output format. I thought this was the way to set the output for `cmd.exe` to `UTF-8` without permanently changing the user's settings.
It's kind of strange for me too, because `go` should only be outputting in `UTF-8`. I was doing some early experimentation with things, so I'll go back and double check if I need to retain this or not.

Reply:
Okay, so I just confirmed that i do need the `Text.Encoding` to ensure output is UTF-8 and not ANSI. But it's not overly complicated that I think we can just build the strings in `durable-task` and simplify this launch portion here.

Reply:
Batch should probably run chcp 65001 if you want utf8 also.   

Unclear if I pipe output to a file what I get but I would think cp 437 /850 by default on American / British systems.  Most code page fun on others.

Reply:
This is the oddity of `go`. For batch files, since it only knows `UTF-8`, no problem, set the `cmd.StdOut` to your file and no problem. For Powershell, however, you need to set `Text.encoding` or else you will get ANSI 🤷

Reply:
batch files know lots of stuff....

https://superuser.com/questions/1408312/control-encoding-of-batch-created-file

Reply:
Pushing the responsibility of encoding to the calling functions in `durable-task`